tests/aircall/test_aircall.py

```diff
@@ -247,3 +302,69 @@ def test_datasource():
         limit=1,
     )
     assert ds.dataset == 'calls'
+
+
+def test_get_status_no_secrets(con, remove_secrets):
+    """
+    It should fail if no secrets are provided
+    """
+    assert con.get_status().status is False
+
+
+def test_get_status_secrets_error(mocker, con):
+    """
+    It should fail if secrets can't be retrieved
+    """
+    mocker.patch(f'{import_path}.OAuth2Connector.get_access_token', side_effect=Exception)
+    assert con.get_status().status is False
+
+
+def test_get_status_api_down(mocker, con):
+    """
+    It should fail if the third-party api is down.
+    """
+    mocker.patch.object(AircallConnector, 'get_access_token', side_effect=HttpError)
+    assert con.get_status().status is False
+
+
+def test_build_authorization_url(mocker, con):
+    """
+    It should proxy OAuth2Connectors methods
+    """
+    mock_oauth2_connector = mocker.Mock(spec=OAuth2Connector)
+    mock_oauth2_connector.client_id = 'test_client_id'
+    mock_oauth2_connector.client_secret = 'test_client_secret'
+    con.__dict__['_oauth2_connector'] = mock_oauth2_connector
+    con.build_authorization_url()
+    mock_oauth2_connector.build_authorization_url.assert_called()
+
+
+def test_specific_retrieve_token(mocker, con):
+    """Check that the AircallConnector way of retrieving access token works"""
+    mock_oauth2_connector = mocker.Mock(spec=OAuth2Connector)
+    mock_oauth2_connector.client_id = 'test_client_id'
+    mock_oauth2_connector.client_secret = 'test_client_secret'
+    con.__dict__['_oauth2_connector'] = mock_oauth2_connector
+    con.__dict__['_oauth2_connector'].authorization_url = 'https://authorization.url'
+    con.retrieve_tokens('toto')
+    mock_oauth2_connector.retrieve_tokens.assert_called_with(
+        'toto', client_id='test_client_id', client_secret='test_client_secret'
+    )
```

Comment:
Also for these ones ^^ I hope we will soon provide you with an easier way to write connectors depending on OAuth2Connector